data/base/endpoint.php

```diff
@@ -330,14 +332,13 @@ public function register_items_route( $methods = WP_REST_Server::READABLE ) {
 	 * @param array $args
 	 *
 	 * @return bool
-	 * @throws \Exception
 	 */
 	public function register_route( $route = '', $methods = WP_REST_Server::READABLE, $callback = null, $args = [] ) {
 		if ( ! in_array( $methods, self::AVAILABLE_METHODS, true ) ) {
-			throw new \Exception( 'Invalid method.' );
+			trigger_error( 'Invalid method.', E_USER_ERROR );
```

Comment:
Why not throw an exception?

Reply:
Will make dependency of '@throws` till the construct will effect also other constructs also will require too many functions to be throw able. this kinda of errors happens only in development stages.